### PR TITLE
Refactor/navigation

### PR DESCRIPTION
**Pull request by weiraneve**
## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## In the scope
- Scaffold使用问题，有多处出现，外层和内层都使用Scaffold的情况。以及Scaffold之后的innerPadding参数配置的问题。
- HomeViewModel 与 icon 之间，MyNavigationBar点击的更换icon逻辑不能落在UI上。
- 去除一些padding配置，改用spacer。
- 修改enum类 HomeTab的名字
- 对HomeViewModel的点击action进行重构
## Checklist
1. [ ] Makes your submission pass tests?
2. [ ] Have you lint your code locally before submission?
3. [ ] Have you updated the wiki of your change?

## Reference
- Card: https://trello.com/c/fxwgKHSm

### CODE REVIEW

**Comment by weiraneve** [comment]
板子 https://trello.com/c/fxwgKHSm

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 1]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
-                actionIcon = MyIcons.Settings,
-            )
-        }
-    ) { innerPadding ->
-        Modifier.padding(innerPadding)
+    val dispatchAction = viewModel::dispatchAction
+    dispatchAction(ForYouAction.GetTopicsAction(LocalContext.current))
+    val topics = viewModel.forYouState.collectAsState().value.topics
+
+    Column(
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: use spacer instead padding

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 2]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
-        topBar = {
-            MyTopBar(
-                modifier = Modifier.zIndex(-1F),
-                title = R.string.app_title,
-                actionIcon = MyIcons.Settings,
-            )
-        }
-    ) { innerPadding ->
-        Modifier.padding(innerPadding)
+    val dispatchAction = viewModel::dispatchAction
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt:31]: no extract dispatchAction

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 3]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
+
+    Column(
+        modifier = Modifier.padding(top = 30.dp)
+    ) {
+        Text(
+            text = stringResource(R.string.for_you_title),
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 24.dp),
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: use spacer instead padding

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 4]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
+            style = MaterialTheme.typography.titleMedium
+        )
+        Text(
+            text = stringResource(R.string.for_you_subtitle),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 8.dp, start = 16.dp, end = 16.dp),
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: use spacer instead padding and use horizontal

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 5]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.bodyMedium
+        )
 
-        val dispatchAction = viewModel::dispatchAction
-        dispatchAction(ForYouAction.GetTopicsAction(LocalContext.current))
-        val topics = viewModel.forYouState.collectAsState().value.topics
+        TopicSelection(
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: rename section

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 6]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
+            Modifier.padding(bottom = 8.dp),
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: spacer

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 7]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
-            TopicSelection(
-                Modifier.padding(bottom = 8.dp),
-                topics
-            )
-
-            Row(
-                horizontalArrangement = Arrangement.Center,
-                modifier = Modifier.fillMaxWidth()
+                    .padding(horizontal = 40.dp)
+                    .width(364.dp),
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt]: alter

File: app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt [hunk 8]
```diff
@@ -19,79 +17,63 @@ import androidx.compose.ui.platform.LocalContext
...
-        dispatchAction(ForYouAction.GetTopicsAction(LocalContext.current))
-        val topics = viewModel.forYouState.collectAsState().value.topics
+        TopicSelection(
+            Modifier.padding(bottom = 8.dp),
+            topics
+        )
 
-        Column(
-            modifier = Modifier.padding(top = 80.dp)
+        Row(
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/ui/page/ForYouScreen.kt:49]: alter no row

File: app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt [hunk 9]
```diff
@@ -1,27 +1,52 @@
...
     fun dispatchAction(action: HomeAction) {
         when (action) {
-            is HomeAction.PositionChangedAction -> onPositionChanged(action.position)
+            is HomeAction.CurrentTabChangedAction -> currentTabChanged(action.currentTab)
         }
     }
 
+}
+
+enum class HomeTabs(
```
Comment (weiraneve) [inline app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt]: quit s

File: app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt [hunk 10]
```diff
@@ -1,27 +1,89 @@
...
+
 class HomeViewModel : ViewModel() {
 
-    private val _position = MutableStateFlow(HomeTabs.FOR_PAGE)
-    val position = _position.asStateFlow()
+    private val _homeState = MutableStateFlow(HomeState())
+    val homeState = _homeState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
```
Comment (yefeidd) [inline app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt]: please remove the viewModelScope

File: app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt [hunk 11]
```diff
@@ -1,27 +1,89 @@
...
+                tabStates = tabStates,
+                currentTab = findHomeTab(currentTabState.title)
+            )
+        }
     }
 
     fun dispatchAction(action: HomeAction) {
         when (action) {
-            is HomeAction.PositionChangedAction -> onPositionChanged(action.position)
+            is HomeAction.CurrentTabChangedAction -> currentTabChanged(action.currentTabState)
```
Comment (yefeidd) [inline app/src/main/java/com/weiran/mynowinandroid/viewmodel/HomeViewModel.kt]: `currentTabChanged` is a noun, please change this function name.

**Review by yefeidd — APPROVED** [review]
LGTM 🚀

**Pull request merged**